### PR TITLE
feat(frontend, meta): support drop table and mv with indexes

### PR DESCRIPTION
**Pull request by chenzl25**
I hereby agree to the terms of the [Singularity Data, Inc. Contributor License Agreement](https://gist.github.com/skyzh/0663682a70b0edde7ae991492f2314cb#file-s9y_cla).

## What's changed and what's your intention?

***PLEASE DO NOT LEAVE THIS EMPTY !!!***

Please explain **IN DETAIL** what the changes are in this PR and why they are needed:

- Previously, if we want to drop a table or a mv with indexes, we need to drop their indexes first. This PR make it possible to drop table or mv with indexes directly when no other mvs are dependent on them.
- I change the RPC Request of drop materialized source and materialized view so that they can take parameters like `index_ids`.
- I also simplify the drop index logic, because there are no internal tables for indexes.

## Checklist

- [x] I have written necessary rustdoc comments
- [x] I have added necessary unit tests and integration tests
- [x] All checks passed in `./risedev check` (or alias, `./risedev c`)

## Documentation

If your pull request contains user-facing changes, please specify the types of the changes, and create a release note. Otherwise, please feel free to remove this section.

### Types of user-facing changes

Please keep the types that apply to your changes, and remove those that do not apply.

* Installation and deployment 
* Connector (sources & sinks)
* SQL commands, functions, and operators
* RisingWave cluster configuration changes
* Other (please specify in the release note below)

### Release note

Please create a release note for your changes. In the release note, focus on the impact on users, and mention the environment or conditions where the impact may occur.

## Refer to a related PR or issue link (optional)
#5749

### CODE REVIEW

**Review by yezizp2012 — APPROVED** [review]
LGTM

File: src/meta/src/manager/catalog/mod.rs [hunk 1]
```diff
@@ -481,67 +481,117 @@ where
...
+            for (index_id, index_table_id) in index_and_table_ids {
+                let index = Index::select(self.env.meta_store(), &index_id).await?;
+                if let Some(index) = index {
+                    index.delete_in_transaction(&mut transaction)?;
+                    assert_eq!(index_table_id, index.index_table_id);
+
+                    // drop index table
+                    let table = Table::select(self.env.meta_store(), &index_table_id).await?;
+                    if let Some(table) = table {
+                        match database_core.get_ref_count(index_table_id) {
```
Comment (yezizp2012) [inline src/meta/src/manager/catalog/mod.rs:514]: Non-related to this PR, do we support or going to support creating index/mv on index now? 🤔

Comment (chenzl25) [reply src/meta/src/manager/catalog/mod.rs:514]: Create an index on an index is forbidden, however, I think creating a mv on an index is ok (considering the delta join case).

Comment (yezizp2012) [reply src/meta/src/manager/catalog/mod.rs:514]: Okay, then this check is still needed. Good job 👍🏻, I think this PR is LGTM now.

**Pull request merged**